src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -334,94 +334,103 @@ int MPIDI_OFI_get_huge_event(int vni, struct fi_cq_tagged_entry *wc, MPIR_Reques
     size_t bytesLeft, bytesToGet;
     MPIR_FUNC_ENTER;
 
-    void *recv_buf = MPIDI_OFI_REQUEST(recv_elem->localreq, util.iov.iov_base);
+    if (recv_elem->localreq && recv_elem->cur_offset != 0) {    /* If this is true, then the message has a posted
+                                                                 * receive already and we'll be able to find the
+                                                                 * struct describing the transfer. */
 
```

Comment:
I don't think we need this in the main branch. You can try replace the branch with `MPIR_Assert` to verify. In the main branch, we only call `MPIDI_OFI_get_huge_event` after both control and recv are matched.